extensions/workspace/watch/output-formatter.ts

```diff
@@ -0,0 +1,40 @@
+import chalk from 'chalk';
+
+export const formatWatchPathsSortByComponent = (watchPathsSortByComponent) => {
+ return watchPathsSortByComponent.reduce((outputString, watchPath) => (
+    outputString + 
+    `${chalk.green('√')} SUCCESS\t${watchPath.componentId}\n` + 
+    `\t - ${watchPath.absPaths?.join('\n')}\n\n`
+  ), ` ${chalk.underline('STATUS\t\tCOMPONENT ID')}\n`) 
+}
+
+export const formatCompileResults = (compileResults, verbose) => (
+  output(compileResults, verbose)
+)
+
+const output = (compileResults, verbose) => {
+
+  return compileResults.map(compileResult => ({
+    extensionId: compileResult.extensionId,
+    resultsForExtension: compileResult.results?.results?.map(resultForExtension => ({
+      component: resultForExtension.component,
+      componentFilesAsString: verboseComponentFilesArrayToString(resultForExtension.buildResults)
+    }))
+  })).reduce((outputString, compileResult) => (
+    outputString + `${resultsForExtensionArrayToString(compileResult.resultsForExtension, verbose)}`
+  ), ` ${chalk.underline('STATUS\t\tCOMPONENT ID')}`)
+}
+
+const resultsForExtensionArrayToString = (resultsForExtension, verbose) => {
+  return resultsForExtension.reduce((outputString, resultForExtension) => (
```

Comment:
probably better to just use a map than in the end do `.join(\n)`

Reply:
Can you please give a code sample of what you mean?

Reply:
?